test/end2end_test.go

```diff
@@ -4238,6 +4243,73 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestClientResourceExhaustedCancelFullDuplex(t *testing.T) {
+	defer leakcheck.Check(t)
+	for _, e := range listTestEnv() {
+		if e.httpHandler {
+			// httpHanlder write won't be blocked on flow control window.
+			continue
+		}
+		testClientResourceExhaustedCancelFullDuplex(t, e)
+	}
+}
+
+func testClientResourceExhaustedCancelFullDuplex(t *testing.T, e env) {
+	te := newTest(t, e)
+	recvErr := make(chan error, 1)
+	timeoutErr := errors.New("1s timeout reached")
+	ts := &funcServer{fullDuplexCall: func(stream testpb.TestService_FullDuplexCallServer) error {
+		_, err := stream.Recv()
+		if err != nil {
+			t.Fatalf("expecting <nil> error, got %v", err)
+		}
+		// create a payload that's larger than the default flow control window.
```

Comment:
You can actually send a smaller payload in a loop and get the same effect.  That way no matter how big the flow control window is, you're guaranteed to get stuck eventually